model_templates/unstructured_with_mlops_reporting/custom.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2023 DataRobot, Inc. and its affiliates.
+# All rights reserved.
+# This is proprietary source code of DataRobot, Inc. and its affiliates.
+# Released under the terms of DataRobot Tool and Utility Agreement.
+
+"""
+This is an example for unstructured custom inference model with MLOps reporting. It is actually
+a binary model, which was defined in DataRobot as `Unstructured (Binary)`, just to demonstrate
+the usage of the `mlops` instance.
+The
+"""
+
+import pickle
+import time
+from pathlib import Path
+import sys
+
+import pandas as pd
+import tempfile
+
+
+def load_model(input_dir):
+    model_path = str(Path(input_dir) / "model.pkl")
+    print(f"Loading model: {model_path}")
+    return pickle.load(open(model_path, "rb"))
+
+
+def score_unstructured(model, data, query, **kwargs):
+    print(f"Model: {model} ", flush=True)
+    print(f"Incoming data type: {type(data)}", flush=True)
+    print(f"Incoming kwargs: {kwargs}", flush=True)
+    print(f"Incoming query params: {query}", flush=True)
+
+    mlops = kwargs.get("mlops")
+
+    df = _get_prediction_request_dataframe(data)
+
+    association_ids = df.iloc[:, 0].tolist()
+    request_array = df.to_numpy()
+
+    # make predictions
+    start_time = time.time()
+    predictions_array = model.predict_proba(request_array)
+    end_time = time.time()
+
+    if mlops:
```

Comment:
Users might ask what are the conditions under which mlops is not available and say "hey, having this mlops client is mandatory, i would rather just assert that it always exists, should I still write if statement?"

Reply:
So, do you suggest to add a comment? or something else?

Reply:
Do we have to be checking "if mlops" if this is for sure an unstructured model with mlops reporting?

Reply:
If the FF is disabled, there will be no `mlops`.

Reply:
Okay, I see. I dont have a strong solution here.

Reply:
I can add a comment and specify the required FF that must be enabled.